File: coding/03rd December 2025/Number of Good Pairs.java
//Number of Good Pairs
class Solution 
{
    public int numIdenticalPairs(int[] nums) 
    {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int ele : nums)
        {
            if(map.containsKey(ele))
                map.put(ele, map.get(ele) + 1);
            else
                map.put(ele, 1);
        }
        int res = 0;
        for(int key : map.keySet())
        {
            int n = map.get(key);
            res += (n * (n - 1)) / 2;
        }
        return res;
    }
}
/*
class Solution 
{
    public int numIdenticalPairs(int[] nums) 
    {
        int count=0;
        for(int i=0;i<nums.length-1;i++)
        {
            for(int j=i+1;j<nums.length;j++)
            {
                if(nums[i]==nums[j])
                    count++;
            }
        }
        return count;
    }
}
*/

File: Coding in Python/23rd February 2026/Number of Good Pairs.py
#Number of Good Pairs
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        dict1 = {}
        result = 0
        for ele in nums:
            if ele in dict1:
                dict1[ele] += 1
            else:
                dict1[ele] = 1
        for k in dict1:
            v = dict1[k]
            result += (v * (v - 1)) // 2
        return result
